src/aggregate-release-notes.js

```diff
@@ -86,92 +111,118 @@ function getUpdatedDependencies(prevVer, currentVer) {
+  ));
+
+  // sort in ascending order by ref
+  return tags
+    .filter((t, i) => i <= upperBound)
+    .sort((v1, v2) => semver.gt(v1.name, v2.name));
 }
 
-async function getAllReleaseNotes(updatedDependencies, options) {
-  const { owner, provider } = options;
-  const matchingTags = [];
+async function getAllReleaseNotes(updatedDependencies, { owner, repo, provider }) {
+  const matchingTags = {};
   let releaseNotes = {};
   for (let key in updatedDependencies) {
     try {
-      matchingTags[key] = await matchTags(key, updatedDependencies[key], options);
-
+      matchingTags[key] = await matchTags(key, updatedDependencies[key], { owner, repo, provider });
     } catch(err) {
       console.error('Error in getAllReleaseNotes', err);
     }
   }
 
-  await Promise.all(Object.keys(matchingTags).map(async function(packageName, index) {
-    releaseNotes[packageName] = await Promise.all(matchingTags[packageName].map(async function(taggedRelease) {
-      let version = taggedRelease.name;
+  // matchingTags :: {<packageName>: [{name: <version>, commit, zipball_url, ...}, ...]}
+  await Promise.all(Object.keys(matchingTags).map(async function(packageName) {
+    // At this step keep changes grouped by the package name:
+    releaseNotes[packageName] = await Promise.all(matchingTags[packageName].map(async function(taggedRelease, index) {
+      // The 1st item is the prevVer that should be excluded (it is just used for the next one to get its prev):
+      if (index === 0) {
+        return;
+      }
+      const version = taggedRelease.name;
+      const currentReleaseSha = taggedRelease.commit.sha;
+      const previousRelease = index > 0 && matchingTags[packageName][index - 1].name;
+      const previousReleaseSha = index > 0 && matchingTags[packageName][index - 1].commit.sha;
       let title = '';
       let body = '';
+      let htmlUrl = '';
 
       try {
         let release = await provider.getReleaseByTag(owner, packageName, version);
 
         if (release.data.name) {
           title = release.data.name;
           body = release.data.body;
+          htmlUrl = release.data.html_url;
         }
 
       } catch(err) {
         // console.error(`${package} ${version}: getReleaseByTag Error Code ${err.code}: ${err.message} `);
       }
 
-      return `[${packageName} ${version}${title ? ' - ' + title : ''}](https://github.com/canjs/${packageName}/releases/tag/${version})${body ? '\n' + body : ''}`;
+      const type = semver.patch(version) !== 0 && 'patch'
+        || semver.minor(version) && 'minor'
+        || semver.major(version) && 'major';
+
+      return {
+        packageName,
+        version,
+        type,
+        previousRelease,
+        currentRelease: version,
+        previousReleaseSha,
+        currentReleaseSha,
+        htmlUrl,
+        title,
+        body
+      };
     }));
+
+    // Filter out unnecessary 1st item:
+    releaseNotes[packageName].shift();
```

Comment:
I think you should move this earlier. I would add a `.then` to the `releaseNotes[packageName] = await Promise.all(...)`.

Reply:
I think this would complicate readability. Do u mind if we keep it as is?